### PR TITLE
feat(deploy-config): adds new module @sap-ux/abap-deploy-config-sub-generator

### PR DESCRIPTION
**Pull request by cianmSAP**
#2526

Adds new module `@sap-ux/abap-deploy-config-sub-generator`

### CODE REVIEW

File: packages/abap-deploy-config-sub-generator/src/app/types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export interface AbapDeployConfigOptions extends AbapDeployConfigAnswersInternal {
+    vscode?: unknown;
+    appWizard?: AppWizard;
+    launchDeployConfigAsSubGenerator?: boolean;
+    launchStandaloneFromYui?: boolean;
+    destinationRoot?: string;
+    projectPath?: string;
+    projectName?: string;
+    connectedSystem?: ConnectedSystem;
+    base?: string;
```
Comment (IainSAP) [inline packages/abap-deploy-config-sub-generator/src/app/types.ts:15]: What is this? Could we please add some docs on the options as these are what consumers need to pass in.

File: packages/abap-deploy-config-sub-generator/src/app/index.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    }
+
+    public async end(): Promise<void> {
+        if (this.abort || this.answers.overwrite === false) {
+            return;
+        }
+
+        // Delayed process of deploy configuration generation if integrated with app generator
+        if (this.launchDeployConfigAsSubGenerator) {
+            await this._initializing();
```
Comment (IainSAP) [inline packages/abap-deploy-config-sub-generator/src/app/index.ts:288]: Not sure why we are initing and writing in the end phase, writing should be done in memory during the writing phase only.

File: packages/abap-deploy-config-sub-generator/src/app/types.ts (hunk 1 above)
Comment (cianmSAP) [reply packages/abap-deploy-config-sub-generator/src/app/types.ts:15]: the `base` file e.g `ui5.yaml`. I didn't like the name either but I can't update this as there is existing documentation to add this via ui5 tooling and command line. 
I will add comments

File: packages/abap-deploy-config-sub-generator/src/app/questions.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    destination?: Destination,
+    backendSystem?: BackendSystem,
+    existingAbapDeployTask?: AbapDeployConfig,
+    backendConfig?: FioriToolsProxyConfigBackend
+): AbapTarget {
+    let url, scp, client, destinationName, authenticationType;
+
+    if (isAppStudio() && destination) {
+        // the destination used during app generation
+        destinationName = destination.Name;
```
Comment (longieirl) [inline packages/abap-deploy-config-sub-generator/src/app/questions.ts:34]: Do we not need any other details here?

Comment (cianmSAP) [reply packages/abap-deploy-config-sub-generator/src/app/questions.ts:34]: Nope, the `AbapTarget` for destination only requires the name 

https://github.com/SAP/open-ux-tools/blob/13f99e6bbef880bb8b803f1e084a07d9c37b80e9/packages/system-access/src/types.ts#L14

**Review by devinea — APPROVED** [review]
New module setup looks good. 
`@sap-ux/abap-deploy-config-sub-generator`

File: packages/abap-deploy-config-sub-generator/src/app/types.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    launchStandaloneFromYui?: boolean;
+    /**
+     * Path to the project
+     */
+    destinationRoot?: string;
+    /**
+     * Path to the project root
+     */
+    projectPath?: string;
```
Comment (IainSAP) [inline packages/abap-deploy-config-sub-generator/src/app/types.ts:30]: Whats the difference between `destinationRoot` //Path to the project and `projectPath` // Path to the project root

File: packages/abap-deploy-config-sub-generator/src/app/types.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Path to the project name
```
Comment (IainSAP) [inline packages/abap-deploy-config-sub-generator/src/app/types.ts:32]: Typo // Path to the project name

**Review by IainSAP — COMMENTED** [review]
Couple of minor comments re. ambigous tsdoc

File: packages/abap-deploy-config-sub-generator/src/app/types.ts (hunk 4 above)
Comment (cianmSAP) [reply packages/abap-deploy-config-sub-generator/src/app/types.ts:26]: I've consolidated these options into `projectPath`.

**Review by longieirl — APPROVED** [review]
Changes look good.
All comments addressed.
Test coverage is good.

**Review by IainSAP — APPROVED** [review]
Thanks @cianmSAP 

- Comments addressed
- Not tested locally

**Review by devinea — APPROVED** [review]
Re-approving.

**Pull request merged**